src/librustdoc/html/render/mod.rs

```diff
@@ -1850,8 +1850,14 @@ fn document_full(w: &mut Buffer, item: &clean::Item, cx: &Context, prefix: &str,
     }
 }
 
-fn document_stability(w: &mut Buffer, cx: &Context, item: &clean::Item, is_hidden: bool) {
-    let stabilities = short_stability(item, cx);
+fn document_stability(
+    w: &mut Buffer,
+    cx: &Context,
+    item: &clean::Item,
+    is_hidden: bool,
+    parent: Option<&clean::Item>,
```

Comment:
The `cfg` in that case is pre-merged, if you look at the second screenshot above the `feature="futures-io"` cfg is coming from the `async_compression::futures::bufread` module, and the `feature="brotli"` cfg is coming from the `BrotliDecoder`, and these are merged into a single `all(feature="futures-io", feature="brotli")` cfg on `BrotliDecoder`. So for simple cases like this, the existing code merging these features down will handle nesting well (and I have yet to see any real code that wants to use non-simple cases).

Reply:
Oh nice! And you actually have test cases for this already, I missed them: https://github.com/rust-lang/rust/pull/77672/files#diff-dbdbc9d891c98a6926bc14ca63c97dfd6a73e8438692533d62eaa73bf32e8eccR169

Reply:
> So for simple cases like this, the existing code merging these features down will handle nesting well (and I have yet to see any real code that wants to use non-simple cases).

That seems fine - if someone needs the complicated cases we can always add them in a follow-up, but I doubt it will come up.